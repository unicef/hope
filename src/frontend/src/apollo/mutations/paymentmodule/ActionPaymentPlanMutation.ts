import { gql } from '@apollo/client';

export const ACTION_PAYMENT_PLAN = gql`
  mutation ActionPP($input: ActionPaymentPlanInput!) {
    actionPaymentPlanMutation(input: $input) {
      paymentPlan {
        id
        status
      }
    }
  }
`;
