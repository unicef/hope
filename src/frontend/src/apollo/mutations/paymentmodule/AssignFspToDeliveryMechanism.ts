import { gql } from '@apollo/client';

export const ASSIGN_FSP_TO_DELIVERY_MECHANISM = gql`
  mutation AssignFspToDeliveryMech($input: AssignFspToDeliveryMechanismInput!) {
    assignFspToDeliveryMechanism(input: $input) {
      paymentPlan {
        id
        deliveryMechanisms {
          id
          name
          fsp {
            id
            name
          }
        }
      }
    }
  }
`;
