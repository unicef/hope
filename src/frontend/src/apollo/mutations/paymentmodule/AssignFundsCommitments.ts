import { gql } from '@apollo/client';

export const ASSIGN_FUNDS_COMMITMENTS = gql`
  mutation AssignFundsCommitmentsPaymentPlan(
    $paymentPlanId: ID!
    $fundCommitmentItemsIds: [String!]!
  ) {
    assignFundsCommitments(
      paymentPlanId: $paymentPlanId
      fundCommitmentItemsIds: $fundCommitmentItemsIds
    ) {
      paymentPlan {
        id
        availableFundsCommitments {
          fundsCommitmentNumber
          fundsCommitmentItems {
            paymentPlan {
              id
              name
            }
            fundsCommitmentItem
            recSerialNumber
          }
        }
      }
    }
  }
`;
