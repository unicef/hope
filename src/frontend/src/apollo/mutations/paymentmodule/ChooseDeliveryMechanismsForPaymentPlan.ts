import { gql } from '@apollo/client';

export const CHOOSE_DELIVERY_MECHANISMS_FOR_PAYMENT_PLAN = gql`
  mutation ChooseDeliveryMechForPaymentPlan(
    $input: ChooseDeliveryMechanismsForPaymentPlanInput!
  ) {
    chooseDeliveryMechanismsForPaymentPlan(input: $input) {
      paymentPlan {
        id
        deliveryMechanisms {
          id
          name
          fsp {
            id
            name
          }
        }
      }
    }
  }
`;
