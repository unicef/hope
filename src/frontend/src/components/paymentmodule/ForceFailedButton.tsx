import { Box, Button, DialogContent, DialogTitle } from '@mui/material';
import * as React from 'react';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Dialog } from '@containers/dialogs/Dialog';
import { DialogActions } from '@containers/dialogs/DialogActions';
import { DialogContainer } from '@containers/dialogs/DialogContainer';
import { DialogFooter } from '@containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '@containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '@hooks/useSnackBar';
import { useMarkPayAsFailedMutation } from '@generated/graphql';

export interface ForceFailedButtonProps {
  paymentId: string;
  disabled?: boolean;
}
export function ForceFailedButton({
  paymentId,
  disabled = false,
}: ForceFailedButtonProps): React.ReactElement {
  const { t } = useTranslation();
  const [isOpenModal, setOpenModal] = useState(false);
  const { showMessage } = useSnackbar();
  const [mutate, { loading }] = useMarkPayAsFailedMutation();

  const submit = async (): Promise<void> => {
    try {
      await mutate({
        variables: {
          paymentId,
        },
      });
      setOpenModal(false);
      showMessage(t('Payment has been marked as failed.'));
    } catch (e) {
      e.graphQLErrors.map((x) => showMessage(x.message));
    }
  };

  return (
    <Box>
      <Box p={2}>
        <Button
          color="primary"
          variant="contained"
          onClick={() => setOpenModal(true)}
          data-cy="button-mark-as-failed"
          disabled={disabled}
        >
          {t('Mark as failed')}
        </Button>
      </Box>
      <Dialog
        open={isOpenModal}
        onClose={() => setOpenModal(false)}
        scroll="paper"
        aria-labelledby="form-dialog-title"
        maxWidth="md"
      >
        <DialogTitleWrapper>
          <DialogTitle>{t('Mark as failed')}</DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              {t('Are you sure you would like to mark payment as failed?')}
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setOpenModal(false)}>{t('CANCEL')}</Button>
            <Button
              type="submit"
              color="primary"
              variant="contained"
              onClick={() => submit()}
              data-cy="button-submit"
              disabled={loading}
            >
              {t('Mark as failed')}
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </Box>
  );
}
