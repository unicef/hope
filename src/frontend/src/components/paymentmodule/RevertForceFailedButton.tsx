import { Box, Button, DialogContent, DialogTitle } from '@mui/material';
import * as React from 'react';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Field, Form, Formik } from 'formik';
import * as Yup from 'yup';
import CalendarTodayRoundedIcon from '@mui/icons-material/CalendarTodayRounded';
import { Dialog } from '@containers/dialogs/Dialog';
import { DialogActions } from '@containers/dialogs/DialogActions';
import { DialogFooter } from '@containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '@containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '@hooks/useSnackBar';
import { FormikTextField } from '@shared/Formik/FormikTextField';
import { FormikDateField } from '@shared/Formik/FormikDateField';
import { useRevertMarkPayAsFailedMutation } from '@generated/graphql';

export interface RevertForceFailedButtonProps {
  paymentId: string;
  disabled?: boolean;
}
export function RevertForceFailedButton({
  paymentId,
  disabled = false,
}: RevertForceFailedButtonProps): React.ReactElement {
  const { t } = useTranslation();
  const [isOpenModal, setOpenModal] = useState(false);
  const { showMessage } = useSnackbar();
  const [mutate, { loading }] = useRevertMarkPayAsFailedMutation();

  const validationSchema = Yup.object().shape({
    deliveredQuantity: Yup.number()
      .min(0)
      .max(99999999, t('Number is too big')),
    deliveryDate: Yup.date().required(t('Delivery date is required')),
  });

  const submit = async (values, { resetForm }): Promise<void> => {
    try {
      await mutate({
        variables: {
          paymentId,
          deliveredQuantity: values.deliveredQuantity,
          deliveryDate: values.deliveryDate,
        },
      });
      setOpenModal(false);
      showMessage(t('Force failed has been reverted.'));
      resetForm();
    } catch (e) {
      e.graphQLErrors.map((x) => showMessage(x.message));
    }
  };

  return (
    <Box>
      <Box p={2}>
        <Button
          color="primary"
          variant="contained"
          onClick={() => setOpenModal(true)}
          data-cy="button-revert-mark-as-failed"
          disabled={disabled}
        >
          {t('Revert mark as failed')}
        </Button>
      </Box>
      <Formik
        initialValues={{
          deliveredQuantity: 0,
          deliveryDate: '',
        }}
        validationSchema={validationSchema}
        onSubmit={submit}
      >
        {({ submitForm, resetForm }) => (
          <Dialog
            open={isOpenModal}
            onClose={() => setOpenModal(false)}
            scroll="paper"
            aria-labelledby="form-dialog-title"
            maxWidth="md"
          >
            <DialogTitleWrapper>
              <DialogTitle>{t('Revert mark as failed')}</DialogTitle>
            </DialogTitleWrapper>
            <DialogContent>
              <Form>
                <Field
                  name="deliveredQuantity"
                  label={t('Delivered Quantity')}
                  type="number"
                  required
                  fullWidth
                  variant="outlined"
                  component={FormikTextField}
                />
                <Field
                  name="deliveryDate"
                  label={t('Delivery Date')}
                  component={FormikDateField}
                  required
                  fullWidth
                  decoratorEnd={<CalendarTodayRoundedIcon color="disabled" />}
                />
              </Form>
            </DialogContent>
            <DialogFooter>
              <DialogActions>
                <Button
                  onClick={() => {
                    setOpenModal(false);
                    resetForm();
                  }}
                >
                  {t('CANCEL')}
                </Button>
                <Button
                  type="submit"
                  color="primary"
                  variant="contained"
                  onClick={submitForm}
                  data-cy="button-submit"
                  disabled={loading}
                >
                  {t('Revert mark as failed')}
                </Button>
              </DialogActions>
            </DialogFooter>
          </Dialog>
        )}
      </Formik>
    </Box>
  );
}
