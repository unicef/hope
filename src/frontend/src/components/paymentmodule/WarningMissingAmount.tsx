import { Box } from '@mui/material';
import WarningIcon from '@mui/icons-material/Warning';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { ReactElement } from 'react';

const WarnIcon = styled(WarningIcon)`
  color: ${({ theme }) => theme.hctPalette.orange};
`;

interface WarningMissingAmountProps {
  amount: number;
  currency: string;
}

export function WarningMissingAmount({
  amount,
  currency,
}: WarningMissingAmountProps): ReactElement {
  const { t } = useTranslation();

  return (
    <Box display="flex" alignItems="center">
      <WarnIcon />
      {t('Missing')} {amount} {currency}
    </Box>
  );
}
