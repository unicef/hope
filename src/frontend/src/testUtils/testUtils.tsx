import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import { Formik } from 'formik';
import noop from 'lodash/noop';
import { TestProviders } from './testProviders';
import { ProgramStatus } from '@generated/graphql';
import { MockLink } from '@apollo/client/testing';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(
    <>
      <BrowserRouter>
        <Formik initialValues={{}} onSubmit={noop}>
          {ui}
        </Formik>
      </BrowserRouter>
    </>,
    {
      wrapper: TestProviders,
      ...options,
    },
  );

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export class ApolloLoadingLink extends MockLink {
  constructor() {
    super([]);
  }

  request() {
    return null;
  }
}

export const fakeContextProgram = {
  selectedProgram: {
    id: '1',
    name: 'someName',
    status: ProgramStatus.Active,
    pduFields: null,
    dataCollectingType: {
      id: '1',
      householdFiltersAvailable: true,
      individualFiltersAvailable: true,
      label: 'data collecting type',
      code: '123',
      type: 'full',
      children: null,
    },
  },
  setSelectedProgram: () => {},
  isActiveProgram: true,
  isSocialDctType: false,
  isStandardDctType: true,
  programHasPdu: false,
};
