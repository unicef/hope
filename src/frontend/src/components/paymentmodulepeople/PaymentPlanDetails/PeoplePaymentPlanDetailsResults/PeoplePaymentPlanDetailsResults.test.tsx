import { MockedProvider } from '@apollo/react-testing';
import { act } from 'react-dom/test-utils';
import wait from 'waait';
import { fakeApolloPaymentPlan } from '../../../../../fixtures/paymentmodule/fakeApolloPaymentPlan';
import { fakeActionPpMutation } from '../../../../../fixtures/paymentmodule/fakeApolloActionPaymentPlanMutation';
import { render } from '../../../../testUtils/testUtils';
import { PeoplePaymentPlanDetailsResults } from './PeoplePaymentPlanDetailsResults';

describe('components/paymentmodule/PaymentPlanDetails/PeoplePaymentPlanDetailsHeader/PaymentPlanDetailsResults', () => {
  it('should render', async () => {
    const { container } = render(
      <MockedProvider addTypename={false} mocks={fakeActionPpMutation}>
        <PeoplePaymentPlanDetailsResults paymentPlan={fakeApolloPaymentPlan} />
      </MockedProvider>,
    );

    await act(() => wait(0)); // wait for the mutation to complete

    expect(container).toMatchSnapshot();
  });
});
