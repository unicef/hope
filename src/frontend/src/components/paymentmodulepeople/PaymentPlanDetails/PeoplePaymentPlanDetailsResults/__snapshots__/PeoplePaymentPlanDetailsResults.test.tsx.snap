// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/PeoplePaymentPlanDetailsHeader/PaymentPlanDetailsResults should render 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 sc-dmyCSP kWtMci css-1ps6pg7-MuiPaper-root"
  >
    <div
      class="sc-eDLKkx kbjSlf"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1j53xw6-MuiTypography-root"
      >
        Results
      </h6>
    </div>
    <div
      class="sc-jTQCzO kXsWaY"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-1efphpz-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              <div
                class="sc-hLQSwg lpoPA-D"
                color="#5F02CF"
              >
                <div>
                  <span
                    class="sc-beySPh gbuZtJ"
                    color="textSecondary"
                  >
                    Female Children
                  </span>
                  <div
                    data-cy="label-Female Children"
                  >
                    <span
                      class="sc-guDLey cyWTlF"
                      color="textSecondary"
                    >
                      0
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              <div
                class="sc-hLQSwg gOSGSG"
                color="#DFCCF5"
              >
                <div>
                  <span
                    class="sc-beySPh gbuZtJ"
                    color="textSecondary"
                  >
                    Female Adults
                  </span>
                  <div
                    data-cy="label-Female Adults"
                  >
                    <span
                      class="sc-guDLey cyWTlF"
                      color="textSecondary"
                    >
                      1
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              <div
                class="sc-hLQSwg epYyAz"
                color="#1D6A64"
              >
                <div>
                  <span
                    class="sc-beySPh gbuZtJ"
                    color="textSecondary"
                  >
                    Male Children
                  </span>
                  <div
                    data-cy="label-Male Children"
                  >
                    <span
                      class="sc-guDLey cyWTlF"
                      color="textSecondary"
                    >
                      0
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              <div
                class="sc-hLQSwg eHLZxv"
                color="#B1E3E0"
              >
                <div>
                  <span
                    class="sc-beySPh gbuZtJ"
                    color="textSecondary"
                  >
                    Male Adults
                  </span>
                  <div
                    data-cy="label-Male Adults"
                  >
                    <span
                      class="sc-guDLey cyWTlF"
                      color="textSecondary"
                    >
                      2
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-1e2bu2o-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
            >
              <div
                class="sc-fsYfdN kEIWSo"
                data-cy="chart-container"
              >
                <canvas
                  height="0"
                  role="img"
                  style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
                  width="0"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-16g5rpk-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              <div
                class="sc-gLLuof chJFWu"
              >
                <div>
                  <span
                    class="sc-beySPh gbuZtJ"
                    color="textSecondary"
                  >
                    Total Number of People
                  </span>
                  <div
                    data-cy="label-Total Number of People"
                  >
                    <span
                      class="sc-guDLey cyWTlF"
                      color="textSecondary"
                    >
                      <div
                        class="sc-iBdnpw laLuMi"
                      >
                        3
                      </div>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
